resources/rest-service/cloudify/migrations/versions/03ad040e6f78_6_1_to_6.2.py

```diff
@@ -139,67 +128,89 @@ def _drop_audit_log_table():
 
 
 # flake8: noqa
-def _create_function_write_audit_log():
+def _create_functions_write_audit_log():
     op.execute("""
-        CREATE OR REPLACE FUNCTION write_audit_log() RETURNS TRIGGER AS $func$
+    CREATE OR REPLACE FUNCTION audit_username() RETURNS TEXT AS $$
+        BEGIN
+            RETURN current_setting('audit.username');
+        EXCEPTION WHEN syntax_error_or_access_rule_violation THEN
+            RETURN NULL;
+        END;
+    $$ LANGUAGE plpgsql;
+
+
+    CREATE OR REPLACE FUNCTION audit_execution_id() RETURNS TEXT AS $$
+        BEGIN
+            RETURN current_setting('audit.execution_id');
+        EXCEPTION WHEN syntax_error_or_access_rule_violation THEN
+            IF (SELECT audit_username() IS NULL) THEN
+                RETURN 'administrative task';
```

Comment:
This might be better as something like 'system' or 'execution'- administrative task sounds like it could've been done by the admin user's request, not anyone else's.

Reply:
I note there's a risk of colliding with a username as well, and I'm not sure we can reasonably insert characters that can't be in a username since we allow external auth providers.

Reply:
There's a good chance username will not be valid if it contained spaces, so, let's put some of those here, like `system task`.